[cleanup] Decouple `FeedbackController` from `FeedbackRepository`

markprudnikov opened the pull request.

markprudnikov committed:
[cleanup] Decouple `FeedbackController` from `FeedbackRepository`

mrchuvyzgalov commented on app-backend/src/main/kotlin/org/example/appbackend/service/FeedbackService.kt:
```diff
@@ -0,0 +1,10 @@
+import org.example.appbackend.entity.FeedbackEntity
+
+interface FeedbackService {
+    fun saveFeedback(feedbackDto: FeedbackDto): FeedbackEntity
```
It'll be better if saveFeedback returns FeedbackDto (but with extra fields). Because Repository layer works with Entity. Service layer works with Dto

mrchuvyzgalov replied:
You can make this function signature: "fun saveFeedback(feedbackDto: CreateFeedbackDto): FeedbackDto"

e2e4b6b7 replied:
Agree

e2e4b6b7 commented on app-backend/src/main/kotlin/org/example/appbackend/controller/FeedbackController.kt:
```diff
@@ -2,27 +2,24 @@ package org.example.appbackend.controller
 ) {
 
     @PostMapping("/feedback")
     fun postFeedback(@RequestBody feedbackDto: FeedbackDto): FeedbackEntity {
```
Actually here is the issue, controller have to return DTO, not entity

markprudnikov committed:
[refactoring] Return `FeedbackDTO` When Posting Feedback

markprudnikov committed:
[refactoring] Introduce `PostFeedbackDTO`

Currently, they're identical, but this would allow us to adjust the data we transfer in DTOs in the future.

e2e4b6b7 reviewed (approved):
Now it is great!
